### ISSUE TITLE
Semantic Segmentation

### ISSUE THREAD

**Issue by makhalid1999** [apple/ml-hypersim#70]
I am trying to replicate the semantic segmentation trained on HyperSim dataset as mentioned in the paper. I am using a different network and a small subset from the dataset for starters. There seems to be a huge class imbalance which causes most of the classes to be labeled as 0 in prediction, whilst achieving a 55%+ accuracy, which would in turn be useless when fine-tuning on NYUv2. I wanted to know if it is possible for you to inform me if you used something to ameliorate this effect. I have used simple  cross entropy and focal losses with different parameters but does not seem to help.

**Comment by mikeroberts3000** [comment]
- In the interest of getting the simplest possible thing working first, I recommend following the standard best-practices for starting machine learning projects (e.g., as described [here](https://x.com/MattNiessner/status/1441027241870118913) and [here](https://youtu.be/O02l_GSFh4E?si=JhH3DwlYbLgKYd2O)).

- If you've already done all that, I recommend using more of our dataset, and trying the training recipe we describe in the paper so you have some idea what to expect.

- I agree that there is a severe class imbalance in Hypersim (similar to the class imbalance in other indoor scene datasets, and in my actual home). But we didn't do anything special to account for that in our experiments. So you should be able to at least roughly match the performance we report in our paper, even though there is a pronounced class imbalance.
